graylog2-server/src/main/java/org/graylog2/rest/resources/streams/rules/StreamRuleResource.java

```diff
@@ -192,4 +195,19 @@ public void delete(@ApiParam(name = "streamid", value = "The stream id this new
             throw new NotFoundException();
         }
     }
+
+    @GET
+    @Path("/types")
+    @Timed
+    @ApiOperation(value = "Get all available stream types")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<Map<String, Object>> types(@ApiParam(name = "streamid", value = "The stream id this new rule belongs to.", required = true)
```

Comment:
Please don't use `Map<String, Object>` as return type for JAX-RS resource methods. This doesn't play well with Swagger and isn't type-safe in any way.